go/tasks/plugins/array/awsbatch/executor.go

```diff
@@ -108,16 +108,17 @@ func (e Executor) Handle(ctx context.Context, tCtx core.TaskExecutionContext) (c
 	// Always attempt to augment phase with task logs.
 	var logLinks []*idlCore.TaskLog
 	var externalResources []*core.ExternalResource
-	switch p {
-	case arrayCore.PhaseStart:
+
+	np, _ := pluginState.GetPhase()
+	if p == arrayCore.PhaseStart && np != arrayCore.PhaseStart {
```

Comment:
This is added logic to hande the new scenario where we stay in PhaseStart for multiple cycles right? i.e., this only runs InitializeExternalResources if something above moves us out of PhaseStart?

Reply:
Correct, the progression of map task plugin state will always start with PhaseStart and transition to PhasePreLaunch / PhaseLaunch to progress. The PhaseStart status continues until the cache lookup is completed (performed in the background). So we basically wait to initialize subtask status' (with admin to track in the UI) until the cache lookups are completed. We need the cache lookups to correctly report which subtasks are read from cache and which are executing.